content/git/concepts/checkout/checkout.md

```diff
@@ -0,0 +1,71 @@
+---
+Title: 'Checkout'
+Description: 'The git checkout command switches branches or restores files in the working directory to a specific state.'
+Subjects: 
+  - 'Bash/Shell'
+  - 'Developer Tools'
+Tags:
+  - 'Git'
+  - 'GitHub'
+CatalogContent:
+  - 'learn-the-command-line'
+  - 'learn-git'
+  - 'learn-github-best-practices'
+---
+
+The git **`checkout`** command switches, create and restore branches in the working directory to a specific state. 
+
+It is also possible to checkout to a specific commit without changing branches.
+
+## Syntax
+
+Checkout with branches:
+```pseudo
+git checkout [options] <branch-name>
+```
+Checkout with a specific commit:
+```pseudo
+git checkout <commit-hash>
+```
+## Switch to an existing branch
+
+The following command will switch to an already existing branch, created previously with the [git branch](https://www.codecademy.com/resources/docs/git/branch) command:
+```pseudo
+git checkout existing-branch
+```
+Please, note that from Git 2.23, the new specific `git switch` command has been introduced to switch branches, making it clearer and safer than `git checkout` because it avoids the ambiguity of the latter's multi-purpose nature:
+```pseudo
+git switch existing-branch
+```
+
+## Create and switch to a new branch
+
+It is possible to create and switch to a new branch with a single command, using the `-b` option:
+```pseudo
+git checkout -b new-branch
+```
+or, alternatively, from Git 2.23:
+```pseudo
+git switch -c new-branch
+```
+
+## Restore a file from a specific commit
+
+Restore a file from a specific commit using its hash:
+```pseudo
+git checkout <commit-hash> -- example.txt
+```
+Please, note that from Git 2.23, the new specific `git restore` command has been introduced to restore files from a specific commit:
+```pseudo
+git restore --source <commit-hash> -- example.txt
+```
+
+## Examine a Previous Commit
+Temporarily move to a specific commit without changing branches. This status is called `detached HEAD state`:
+```pseudo
+git checkout <commit-hash>
+```
+The detached HEAD state allow to:
+- Examine the state of the repository at that specified commit.
+- Create new branches if the develop needs to start from that point.
```

Comment:
There's possibly a typo here, not sure if you meant `development` or `developer` here. Please check -

```suggestion
- Create new branches if the developer needs to start from that point.
```